### ISSUE TITLE
Headless ?

### ISSUE THREAD

**Issue by cyrilchapon** [browserslist/browserslist-ga#4]
Is there a way to use, say, env variables to update the file in pure headless mode ?

This would allow hooking it in our CI pipeline, thus updating stats every push in the process

**Comment by dmfrancisco** [comment]
Thanks for opening this issue 👍 The short answer is **no**.

This is something that I also need and I'm planning to do (but please feel free to send a PR if you end working on it!). I'm curious how you were planning on integrating with your CI. Any chance you could share your thoughts?

In many CI services you have access to the git history and so with support for headless mode you could run `browserslist-ga` and commit back the changes (if those changes were relevant enough) to your version control (similarly to what [`greenkeeper-lockfile`](https://github.com/greenkeeperio/greenkeeper-lockfile) does). Is this the kind of setup you were thinking of, or is your intent to skip adding `browserslist-stats.json` to your version control and have it compiled upon deliver?

**Comment by cyrilchapon** [comment]
> I'm curious how you were planning on integrating with your CI. Any chance you could share your thoughts?

> In many CI services you have access to the git history and so with support for headless mode you could run browserslist-ga and commit back the changes

Well I'd go with a KISS approach, `gitignore browserslist-stats.json` and run

package.json
```
[...],
scripts: {
}
[...]
```

**Comment by cyrilchapon** [comment]
> I'm curious how you were planning on integrating with your CI. Any chance you could share your thoughts?

> In many CI services you have access to the git history and so with support for headless mode you could run browserslist-ga and commit back the changes

Well I'd go with a KISS approach, `gitignore browserslist-stats.json` and run

package.json
```
{
  [...],
  "scripts": {
    "build": "browserlist-ga && [WHATEVER BUILD CMD]"
  }
  [...]
}
```

:D

**Comment by dmfrancisco** [comment]
That makes sense, thanks for sharing 👍 I believe there are some benefits in having this file under version control, but on the other hand as you mentioned there's the risk that the setup becomes too complicated. But either way being able to run `browserslist-ga` in non-interactive mode is necessary and so it's something I want to work on soon. I'll report back once I have more news!

**Comment by cyrilchapon** [comment]
Good to hear ☺️

I'm full on side projects in here, I believe I won't have a chance to PR something.

Thinking about it, (versioning stuff), I thought the following :

Without browserslist-ga, when you use for example babel-preset-env, the build may be different everytime anyway  if you have something like ">1%".

Having the ability to rebuild exactly the same stuff sounds fair, but not that familiar after all.

Plus, in a CI env, you'll probably insert your test command **between** build and start, and good practices dictate to test built code instead of source code. So this seems safe to me ☺️

Le 27 mars 2018 à 19:50, David Francisco <notifications@github.com<mailto:notifications@github.com>> a écrit :


That makes sense, thanks for sharing 👍 I believe there are some benefits in having this file under version control, but on the other hand as you mentioned there's the risk that the setup becomes too complicated. But either way being able to run browserslist-ga in non-interactive mode is necessary and so it's something I want to work on soon. I'll report back once I have more news!

**Comment by dmfrancisco** [comment]
I agree 👍 One nitpick regarding `babel-preset-env` and other tools that use Browserslist, is that those queries (like `> 1%`) actually run against a static file inside the [`caniuse-lite`](https://github.com/ben-eb/caniuse-lite) dependency. So if you have a `yarn.lock` or `package-lock.json` in your project, it should use the same data across builds.

**Comment by Hoten** [comment]
I am leaning towards not wanting a headless mode. I am thinking of a case where GA stops reporting data (For whatever reason), this goes unnoticed, and the generated browserslist is missing tons of browsers that very-much ought to be under support. I think having this tool be manually run and audited is important.
